### ISSUE TITLE
[EN][Raid] Majordomo Staghelm: Leaping Flames 

### ISSUE THREAD

**Issue by Psychobird7** [Atlantiss/BugTracker#9453]
**Description**: Majordomo Staghelm's Leaping Flames currently uses an incorrect target selection system. 

**Current behaviour**: Majordomo Staghelm only leaps at targets standing out of his Melee range. 

**Expected behaviour**: Majordomo Staghelm should choose Leap targets based on their specialisation, and target all ranged DPS, casters and healers regardless of position. 

https://www.icy-veins.com/wow/majordomo-staghelm-strategy-guide-normal-heroic 
"6.2.2 Cat Form 
**Majordomo Staghelm chooses targets to leap on based on their class/spec. Thus, healers and ranged DPS can be targeted by a leap regardless of whether they are in melee range or not**, and as a leap in melee is basically an instant wipe, you cannot have ranged DPS or healers staying there to avoid being damaged." 

http://www.wowhead.com/spell=98535/leaping-flames#comments:id=1554896 
"a little late to comment, but no, **the encounter scans your spec and will never attack a melee**. 
If a ranged class comes close he will still pounce them.
personally verified multiple times"

http://www.mmo-champion.com/threads/964817-Majordomo-Staghelm-Heroic-Advice-Needed 
See comments 4 and 12. 
-- "No, because **he leaps at you based on your spec**. Anyone that is in melee range and doesn't need to be has a chance to be leaped on, killing the entirety of the melee." 
-- "Is intended to **leap by spec**, so doesnt matter if your melee range.
All **ranged /healers shouldnt be in melee range**." 

Bonus reason: 
Top guilds had all ranged, casters and healers spread into range. 
Due to the boss's extremely high DPS requirements at the time, it is extremely unlikely that guilds racing progression on this boss would not have tried everything in their power to minimise movement for their casters. 

More to be added soon. 

TL;DR: Majordomo Staghelm's Leaping Flames should choose targets based on their specialisation. 
All ranged/healer specs should be eligible targets regardless of where they're standing.  

**Steps to reproduce the problem**:

1. Enrage Majordomo (normal/heroic, does not matter) 
2. Have one or several casters stay in Melee range to turret DPS 
3. See the boss will not target them with Leaping Flames. 

**Revision**: 5480

**Comment by tomatosz** [comment]
Only one I could doubt in your list are the hpallys, since they benefit from meleeing (or so I've been told, don't play it) so they could potentially be classified as melee in this encounter (?). 
Also, I'm fine with engaging, but would rather not enrage majordomo =D

**Comment by Psychobird7** [comment]
@tomatosz  
Threads/guides do not specify whether HPals are an exception to the targeting rule, but since they do fall under Healer specs, it woud make sense for them to be eligible targets.  
I will obviously dig for more evidence but help is welcome if you can find any.

**Comment by tomatosz** [comment]
Disregard my last comment, checked a hpally pov vid and saw domo jump on him.

**Comment by Ifernoian** [comment]
bump
